filecache/file_cache_source.py

```
@@ -1289,7 +1293,12 @@ def upload(self,
             # For some reason the Google API doesn't let you update this directly
             blob.metadata = blob_metadata
 
-        blob.upload_from_filename(str(local_path))
+        try:
+            blob.upload_from_filename(str(local_path))
+        except Exception as e:
+            raise UploadFailed(
+                f'Failed to upload file to Google Storage: '
+                f'{self._src_prefix_}{sub_path}') from e
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)